crates/cairo-lang-protostar/src/cli.rs

```diff
@@ -0,0 +1,44 @@
+//! Compiles and runs a Cairo program.
+
+use anyhow::{Context};
+use cairo_lang_sierra::ProgramParser;
+use cairo_lang_sierra::program::Program;
+use casm_generator::SierraCasmGenerator;
+use std::fs;
+
+use clap::Parser;
+
+// use casm_generator::SierraCasmGenerator;
+mod casm_generator;
+
+#[derive(Parser, Debug)]
+#[clap(version, verbatim_doc_comment)]
+struct Args {
+    /// The file to compile
+    file: String,
+    /// The output file name (default: stdout).
+    output: Option<String>,
+}
+
+fn main() -> anyhow::Result<()> {
+    let args = Args::parse();
+    
+    let sierra_code = fs::read_to_string(args.file).expect("Could not read file!");
```

Comment:
You are returning `Result` from `main` so you can return errors from it. Context from anyhow provides nice logical trace of the error.

```suggestion
    let sierra_code = fs::read_to_string(args.file).context("Could not read file.")?;
```